File: OptimoveSDK/optimove-sdk/src/test/java/com/optimove/android/gamifywidgetsdk/AndroidBridgeTest.java
package com.optimove.android.gamifywidgetsdk;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.never;
import static org.mockito.Mockito.verify;

import org.junit.Test;

public class AndroidBridgeTest {

    @Test
    public void closeWidget_invokesOnCloseCallback() {
        Runnable onClose = mock(Runnable.class);
        AndroidBridge bridge = new AndroidBridge(onClose, mock(Runnable.class));
        bridge.closeWidget();
        verify(onClose).run();
    }

    @Test
    public void receiveMessage_invokesOnReadyForReadyType() {
        Runnable onReady = mock(Runnable.class);
        AndroidBridge bridge = new AndroidBridge(mock(Runnable.class), onReady);
        bridge.receiveMessage("{\"type\":\"READY\"}");
        verify(onReady).run();
    }

    @Test
    public void receiveMessage_doesNotInvokeOnReadyForOtherTypes() {
        Runnable onReady = mock(Runnable.class);
        AndroidBridge bridge = new AndroidBridge(mock(Runnable.class), onReady);
        bridge.receiveMessage("{\"type\":\"TEST\"}");
        verify(onReady, never()).run();
    }
}

File: OptimoveSDK/gamify-widget-sdk/src/test/java/com/optimove/android/gamifywidgetsdk/AndroidBridgeTest.kt
package com.optimove.android.gamifywidgetsdk

import org.junit.Test
import org.mockito.kotlin.mock
import org.mockito.kotlin.never
import org.mockito.kotlin.verify

class AndroidBridgeTest {

    @Test
    fun `closeWidget invokes onClose callback`() {
        val onClose = mock<() -> Unit>()
        val bridge = AndroidBridge(onClose = onClose, onReady = {})
        bridge.closeWidget()
        verify(onClose).invoke()
    }

    @Test
    fun `receiveMessage invokes onReady for READY type`() {
        val onReady = mock<() -> Unit>()
        val bridge = AndroidBridge(onClose = {}, onReady = onReady)
        bridge.receiveMessage("""{"type":"READY"}""")
        verify(onReady).invoke()
    }

    @Test
    fun `receiveMessage does not invoke onReady for other types`() {
        val onReady = mock<() -> Unit>()
        val bridge = AndroidBridge(onClose = {}, onReady = onReady)
        bridge.receiveMessage("""{"type":"TEST"}""")
        verify(onReady, never()).invoke()
    }

}
